src/audio/pipeline.c

```diff
@@ -441,6 +441,9 @@ void pipeline_cache(struct pipeline *p, struct comp_dev *dev, int cmd)
 	/* execute cache operation on components and buffers */
 	pipeline_comp_cache(dev, &data, dev->params.direction);
 
+	/* execute cache operation on pipeline task */
+	schedule_task_cache(&p->pipe_task, cmd);
```

Comment:
Shouldn't this level of scheduler detail be invisible to callers ? This should be implied when the task is registered, i.e. task can be run on any core.

Reply:
Yes, you're right. I will do it internally in the schedulers. The whole management of shared memory between cores needs to be reimplemented. Currently it's too vulnerable and it doesn't scale too well.